scripts/megastock/write_databricks_to_bigquery_03.py

```diff
@@ -18,52 +25,116 @@
 
 # COMMAND ----------
 
+from itertools import chain
+
 from google.cloud import bigquery
+import pyspark.sql.functions as F
+from pyspark.sql.functions import udf
+from pyspark.sql.types import IntegerType
 
 # COMMAND ----------
 
 N_SAMPLE_TAG = dbutils.widgets.get("n_sample_tag")
 
 # COMMAND ----------
 
-bq_metadata_write_path = 'cube-machine-learning.ds_api_datasets.megastock_metadata'
+# Initialize a BigQuery client
+client = bigquery.Client()
+
+# set up paths to write to 
+bq_project = "cube-machine-learning"
+bq_dataset = "ds_api_datasets"
+bq_megastock_table = 'megastock_combined_baseline'
+bq_write_path = f"{bq_project}.{bq_dataset}.{bq_megastock_table}"
+
+# COMMAND ----------
+
+# read in data
 building_metadata = spark.table(f'ml.megastock.building_metadata_{N_SAMPLE_TAG}')
+building_features = spark.table(f'ml.megastock.building_features_{N_SAMPLE_TAG}')
+
+# COMMAND ----------
+
+# get distinct climate zones
+climate_zone_list = list(map(lambda row: row['ashrae_iecc_climate_zone_2004'], building_metadata.select('ashrae_iecc_climate_zone_2004').distinct().collect()))
 
+# Mapping dictionary
+climate_zone_mapping = {cz: i for i, cz in (enumerate(sorted(climate_zone_list), 1))}
+
+# Define UDF
+cz_mapping_expr = F.create_map([F.lit(x) for x in chain(*climate_zone_mapping.items())])
+
+# Apply UDF to create new column
+building_metadata_with_int_cz = building_metadata.withColumn("climate_zone_int", cz_mapping_expr[F.col('ashrae_iecc_climate_zone_2004')])
 
 # COMMAND ----------
 
-(building_metadata
-    .write
-    .format("bigquery")
-    .mode("overwrite")
-    .option("table", bq_metadata_write_path)
-    .option("temporaryGcsBucket", "the-cube")
-    .save()
+# Add "__m" suffix to all columns in metadata table except "building_id"
+building_metadata_renamed = building_metadata_with_int_cz.select(
+    [F.col(c).alias(f"{c}__m") if c != "building_id" else F.col(c) for c in building_metadata_with_int_cz.columns]
 )
 
 # COMMAND ----------
 
-bq_features_write_path = 'cube-machine-learning.ds_api_datasets.megastock_features'
-building_features = spark.table(f'ml.megastock.building_features_{N_SAMPLE_TAG}')
+# Subset features to baseline only and join the tables on building_id
+combined_df_baseline = building_metadata_renamed.join(
+    building_features.where(F.col('upgrade_id') == 0).drop('upgrade_id'),
+    on="building_id", how="inner")
+
+# COMMAND ----------
 
-(building_features
+# write out table to big query
+(combined_df_baseline
     .write
     .format("bigquery")
     .mode("overwrite")
-    .option("table", bq_features_write_path)
+    .option("table",bq_write_path)
     .option("temporaryGcsBucket", "the-cube")
     .save()
 )
 
-
 # COMMAND ----------
 
-# check that tables are there
-client = bigquery.Client()
+# Drop table
```

Comment:
Is this dropping the table right before you try to read from it?

Reply:
so sorry i had uncomitted code i forgot to push! I'm now so used to working locally that I am rusty in databricks.